plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanel.java

```diff
@@ -141,23 +143,27 @@ private void setupDataBinding() {
   }
 
   private void setupAccountEmailDataBinding(DataBindingContext context) {
-    final IObservableValue accountEmailModel = PojoProperties.value("accountEmail").observe(model);
-    context.bindValue(new AccountSelectorObservableValue(accountSelector), accountEmailModel);
-
-    if (requireValues) {
-      context.addValidationStatusProvider(new FixedMultiValidator() {
-        @Override
-        protected IStatus validate() {
-          String email = (String) accountEmailModel.getValue();
-          // It's possible that no account is selected while a valid email has been saved in the
-          // model (if the corresponding account is signed out), so check the actual selection too.
-          if (email.isEmpty() || accountSelector.getSelectedEmail().isEmpty()) {
-            return ValidationStatus.error(Messages.getString("error.account.missing"));
-          }
-          return ValidationStatus.ok();
+    IValidator checker = new IValidator() {
+      @Override
+      public IStatus validate(Object value /* email */) {
+        if (requireValues && Strings.isNullOrEmpty((String) value)) {
+          return ValidationStatus.error(Messages.getString("error.account.missing"));
         }
-      });
-    }
+        return ValidationStatus.ok();
+      }
+    };
+    UpdateValueStrategy targetToModel = new UpdateValueStrategy().setBeforeSetValidator(checker);
+    UpdateValueStrategy modelToTarget = new UpdateValueStrategy().setBeforeSetValidator(checker);
+    modelToTarget.setConverter(new Converter(String.class, String.class) {
+      @Override
+      public Object convert(Object fromObject /* email */) {
+        return accountSelector.isEmailAvailable((String) fromObject) ? fromObject : null;
```

Comment:
Should we be using the email as an account key then?  If the model stores "", and there are two accounts with blank email, which one should be selected?  Perhaps we should be using our own UID.